WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -1043,7 +1046,13 @@ import WordPressComAnalytics
     /// are met.
     /// - The app must have a internet connection.
     /// - The current time must be greater than the last sync interval.
-    func syncIfAppropriate() {
+    /// - Autosync must not have already occurred.
+    ///
+    /// NOTE:  We want to avoid automatically syncing when the controller is created
+    /// via state restoration until the controller's view is in the window.
+    /// See: https://github.com/wordpress-mobile/WordPress-iOS/issues/6297
+    ///
+    func autoSyncIfAppropriate(viewWillAppear viewWillAppear: Bool) {
```

Comment:
The goal of the patch is to prevent auto syncing unless the stream is the active view controller. This fixes the crash you found by avoiding any deletion or cleanup triggered by the auto sync while content is still being viewed.
You're feedback got me rethinking the approach and I believe I've been able to simplify things and still achieve the goal.
Ready for another look :)